cls documentation (#575)
kyma-project/control-plane: docs/kyma-environment-broker/04-01-cls-architecture.md

---

> title: Cloud Logging Service Architecture

Reviewer: ```suggestion
title: Cloud Logging Service Integration
```
We are not talking about the Architecture of CLS here

type: Architecture
---

> Cloud Logging Service (CLS) is a managed logging service for shipping logs from Runtimes. It uses the Elastic Stack (open source logging platform Elasticsearch, Logstash, Kibana) to store, parse, and visualize the application log data coming from Kubernetes applications.

Reviewer 2: ```suggestion
The Cloud Logging service (CLS) is a managed logging service for shipping logs from various runtimes. It uses the Elastic Stack (open source logging platform Elasticsearch, Logstash, Kibana) to store, parse, and visualize the application log data coming from Kubernetes applications.
```

Reviewer 2: Official [CLS docu](https://pages.github.tools.sap/perfx/cloud-logging-service/consumption/getting-started/) writes it this way.

I think "runtimes" is a normal word, no need to capitalize.

> Cloud Logging Service (CLS) is a managed logging service for shipping logs from Runtimes. It uses the Elastic Stack (open source logging platform Elasticsearch, Logstash, Kibana) to store, parse, and visualize the application log data coming from Kubernetes applications.

Reviewer: It is an SAP specific service and it is using FluentD instead of Logstash and it can handle application logs not only from kubernetes based applications

>  Kyma environment broker provisions a CLS instance for a global account, creates a binding and passes the CLS credentials to the provisioner as SKR overrides so that fluent bit in the SKR can be properly configured to push the logs to the CLS instance. The architecture diagram below explains the complete process

Reviewer 2: [Our docu](https://kyma-project.io/docs/control-plane/kyma-environment-broker/#documentation-content) capitalize this. 
Not sure about the "can be" - is this optional? If not, I'd use "is configured. Or are further user actions needed? Then I suggest "so that you can configure..."
```suggestion
 Kyma Environment Broker provisions a CLS instance for a global account and creates a binding. It passes the CLS credentials to the provisioner as SKR overrides, so that fluent bit in the SKR is properly configured to push the logs to the CLS instance. The following architecture diagram explains the complete process:
```

>  Kyma environment broker provisions a CLS instance for a global account, creates a binding and passes the CLS credentials to the provisioner as SKR overrides so that fluent bit in the SKR can be properly configured to push the logs to the CLS instance. The architecture diagram below explains the complete process

Reviewer: ```suggestion
 Kyma environment broker provisions a CLS instance for a global account, creates a binding per Kyma cluster and passes the CLS credentials to the provisioner as SKR overrides so that fluent bit in the SKR can be properly configured to push the logs to the CLS instance. The architecture diagram below explains the complete process
```

![CLS diagram](./assets/cls-arch.svg)

1. CSI sends the request to KEB to provision a new KEB.

> 2. If there is not CLS instance present for the global account, the it provisions a new instance for CLS for the global account.

Reviewer 2: Who is "it"? KEB? CSI? The arrow goes from CSI to KEB, so I assume CSI provides it.

```suggestion
2. If there is no CLS instance present for the global account, then CSI provisions a new instance for CLS for the global account.
```

> 2. If there is not CLS instance present for the global account, the it provisions a new instance for CLS for the global account.

Reviewer: ```suggestion
2. If there is not CLS instance present yet for the global account, KEB provisions a new instance of CLS for the global account.
```

>     a. If the CLS instance is already provisioned for the global account then KEB would create a new binding. see [3.](3)

Reviewer 2: I assume the two "If..." sentences are alternatives. I suggest to document them as unordered list below one procedure step - either below step 1, or under a description like "T

```suggestion
      - If the CLS instance is already provisioned for the global account, then KEB creates a new binding. see [3.](3)
```

The link to 3 does not work for me, not sure why. Where does it point to? Step 3 of this procedure?

> 3. After the CLS instance is provisioned then we create a binding through which we get the credentials to push logs to the CLS instance.

Reviewer: Remove the push part here, it will confuse
